pkg/controller/utils/gcp/utils.go

```diff
@@ -147,22 +147,43 @@ func filterOutPermissions(permList, filterOutList []string) []string {
 // gcpClient creds have sufficient permissions to perform the actions.
 // Will return true/false indicating whether the permissions are sufficient.
 func CheckPermissionsAgainstPermissionList(gcpClient ccgcp.Client, permList []string, logger log.FieldLogger) (bool, error) {
+
+	if len(permList) == 0 {
+		return true, nil
+	}
+
 	projectName := gcpClient.GetProjectName()
 
 	filteredPermList := filterOutPermissions(permList, invalidPermsForProjectScopedPermissionsTesting)
 
-	permRequest := &cloudresourcemanager.TestIamPermissionsRequest{
-		Permissions: filteredPermList,
-	}
-	permResponse, err := gcpClient.TestIamPermissions(projectName, permRequest)
-	if err != nil {
-		return false, fmt.Errorf("error testing permissions: %v", err)
+	// Split list to only check 100 permissions at a time
+	responses := []*cloudresourcemanager.TestIamPermissionsResponse{}
+	done := false
+	index := 0
+	for !done {
+		upperLimit := len(filteredPermList)
+		if index+100 < upperLimit {
+			upperLimit = index + 100
+		} else {
+			done = true
+		}
+		permRequest := &cloudresourcemanager.TestIamPermissionsRequest{
+			Permissions: filteredPermList[index:upperLimit],
+		}
+		permResponse, err := gcpClient.TestIamPermissions(projectName, permRequest)
+		if err != nil {
+			return false, fmt.Errorf("error testing permissions: %v", err)
+		}
+		responses = append(responses, permResponse)
+		index += 100
 	}
 
 	// check that each perm in our list is actually available
 	permMap := map[string]bool{}
-	for _, perm := range permResponse.Permissions {
-		permMap[perm] = true
+	for _, resp := range responses {
```

Comment:
nit: instead of collecting a slice of responses, I'd rather create `permMap` before the chunk loop and then iterate over each chunk's response inside the loop:

```go
permMap := map[string]bool{}
for i := 0; i < permLen; i += chunkSize {
  ...
  for _, perm := range resp.Permissions {
    permMap[perm] = true
  }
}
```